### PR TITLE
Support timeline background press for creating new events

### PR DESCRIPTION
**Pull request by ethanshar**
- Support timeline background (long) press for creating new events - currently this is a very simple behavior till we have the final UX
- Enable TS in the Timeline example screen 
- Added usage example
- Move logic to a presenter file and add tests

### CODE REVIEW

**Review by lidord-wix — COMMENTED** [review]
Looks good overall, I left a few comments, and here are more general comments:  
- IMO, the background long press should "floor" the time of the created event, for example - pressing on 13:50 should set a new event at 13:30-14:30 and not at 14:00-15:00.
- It doesn't reproduce all the time, but sometimes when I long-press the background it doesn't create a new event 
- There's a small conflict with master to resolve

File: src/timeline/Timeline.tsx [hunk 1]
```diff
@@ -6,22 +6,45 @@ import map from 'lodash/map';
...
+   */
+  eventTapped?: (event: Event) => void;
+  /**
+   * Handle event press
+   */
   onEventPress?: (event: Event) => void;
+  /**
+   * Pass to handle creation of a new event by long press on the timeline background
+   */
+  onEventCreate?: TimelineHoursProps['onBackgroundLongPress'];
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/timeline/Timeline.tsx:37]: I will actually change it a little anyway now that I have the UX guidelines. 
I will have two callbacks 
onLongPress -> this should generate a new event placeholder
onPressOut -> this should invoke the event creation callback 
This is how it works in Calendars apps, so I'll try to mimic that

File: example/src/screens/timelineCalendar.tsx [hunk 2]
```diff
@@ -101,7 +100,24 @@ export default class TimelineCalendarScreen extends Component {
...
   };
 
+  createNewEvent: TimelineProps['onEventCreate'] = (timeString, timeObject) => {
+    const {currentDate, newEvents} = this.state;
+
+    const newEvent = {
+      start: `${currentDate} ${timeString}`,
+      end: `${currentDate} ${(timeObject.hour + 1).toString().padStart(2, '0')}:${timeObject.minutes
+        .toString()
+        .padStart(2, '0')}:00`,
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply example/src/screens/timelineCalendar.tsx]: Fixed

File: src/timeline/Timeline.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/timeline/Timeline.tsx]: Fixed, you can see how I use it now in the example screen

**Comment by ethanshar** [comment]
> Looks good overall, I left a few comments, and here are more general comments:
> 
> * IMO, the background long press should "floor" the time of the created event, for example - pressing on 13:50 should set a new event at 13:30-14:30 and not at 14:00-15:00.
> * It doesn't reproduce all the time, but sometimes when I long-press the background it doesn't create a new event
> * There's a small conflict with master to resolve

- Yes, I agree. I fixed the rounding logic
- It works fine for me, see now with the new creating logic if you still have issues
- Fixed ✔️

File: example/src/screens/timelineCalendar.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (lidord-wix) [reply example/src/screens/timelineCalendar.tsx]: Sorry, I wasn't clear enough.. I meant to add consts like: 
```
const hourString  = (timeObject.hour + 1).toString().padStart(2, '0');
const minutesString = timeObject.minutes.toString().padStart(2, '0');
```

File: src/timeline/Timeline.tsx [hunk 3]
```diff
@@ -34,7 +34,11 @@ export interface TimelineProps {
   /**
    * Pass to handle creation of a new event by long press on the timeline background
    */
-  onEventCreate?: TimelineHoursProps['onBackgroundLongPress'];
+  onBackgroundLongPress?: TimelineHoursProps['onBackgroundLongPress'];
+  /**
+   * Pass to handle creation of a new event by long press out on the timeline background
+   */
+   onBackgroundLongPressOut?: TimelineHoursProps['onBackgroundLongPressOut'];
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/timeline/Timeline.tsx:41]: Fixed (:

File: example/src/screens/timelineCalendar.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ethanshar) [reply example/src/screens/timelineCalendar.tsx]: Cool. Fixed 👍

**Review by lidord-wix — APPROVED** [review]

**Pull request merged**